hazelcast/src/main/java/com/hazelcast/spi/impl/merge/FullMergingEntryHolderImpl.java

```diff
@@ -189,6 +190,11 @@ public long getTtl() {
         return this;
     }
 
+    @Override
+    public void setNode(Node node) {
```

Comment:
Why is a dependency to Node required if a dependency to serialization would suffice?

Reply:
That's just the contract required when implementing `NodeAware`. I don't think there's a more specific one that we can use here, e.g. `SerialisationServiceAware`.

Reply:
It's an internal class and `NodeAware` is the easiest way to get your hands on the `SerializationService`. You cannot retrieve it officially from a `HazelcastInstance` and there is no `SerializationServiceAware`.

Reply:
It also makes the other code look really nice, since we can use the same injection method like for the merge policy:
```java
void merge(MergingValueHolder mergingValue, SplitBrainMergePolicy mergePolicy) {
    serializationService.getManagedContext().initialize(mergingValue);
    serializationService.getManagedContext().initialize(mergePolicy);
    ...
}
```

Reply:
The contract NodeAware is no good. There should not be any dependencies on Node if not needed; Node is a terrible class.

Reply:
I have no objections to introducing `SerializationServiceAware` then, until we introduce dependency injection (or Spring :) )

Reply:
Done...

Reply:
Added the injection to `HazelcastManagedContext` and `HazelcastClientManagedContext`, also added a missing test for the client context. The `SpringManagedContext` needs to be fixed in another PR, since it doesn't even inject `HazelcastInstanceAware` etc.